File: Examples/Python/udp_connection.py
import connection
import helpers
import ximu3

if helpers.yes_or_no("Search for connections?"):
    messages = ximu3.NetworkAnnouncement().get_messages_after_short_delay()

    if not messages:
        raise Exception("No UDP connections available")

    print(f"Found {messages[0].device_name} {messages[0].serial_number}")

    connection_info = messages[0].to_udp_connection_info()

    connection.run(connection_info)
else:
    connection_info = ximu3.UdpConnectionInfo("192.168.1.1", 9000, 8000)  # replace with actual connection info

    connection.run(connection_info)

File: Examples/CSharp/x-IMU3-Examples/Examples/UdpConnection.cs
namespace Ximu3Examples
{
    class UdpConnection : Connection
    {
        public UdpConnection()
        {
            if (Helpers.YesOrNo("Search for connections?"))
            {
                var networkAnnouncement = new Ximu3.NetworkAnnouncement();

                var result = networkAnnouncement.GetResult();

                if (result != Ximu3.CApi.XIMU3_Result.XIMU3_ResultOk)
                {
                    Console.WriteLine("Network announcement failed. " + Ximu3.Helpers.ToString(Ximu3.CApi.XIMU3_result_to_string(result)) + ".");
                    return;
                }

                var messages = networkAnnouncement.GetMessagesAfterShortDelay();

                if (messages.Length == 0)
                {
                    Console.WriteLine("No UDP connections available");
                    return;
                }

                Console.WriteLine("Found " + Ximu3.Helpers.ToString(messages[0].device_name) + " " + Ximu3.Helpers.ToString(messages[0].serial_number));

                var config = new Ximu3.UdpConnectionConfig(messages[0]);

                Run(config);
            }
            else
            {
                var config = new Ximu3.UdpConnectionConfig("192.168.1.1", 9000, 8000); // replace with actual connection config

                Run(config);
            }
        }
    }
}
